ffi/src/rdcleanpath.rs

```diff
@@ -0,0 +1,160 @@
+#[derive(Clone, Copy, Debug)]
+struct RDCleanPathHint;
+
+const RDCLEANPATH_HINT: RDCleanPathHint = RDCleanPathHint;
+
+impl ironrdp::pdu::PduHint for RDCleanPathHint {
+    fn find_size(&self, bytes: &[u8]) -> ironrdp::core::DecodeResult<Option<(bool, usize)>> {
+        match ironrdp::rdclean_path::RDCleanPathPdu::detect(bytes) {
+            ironrdp::rdclean_path::DetectionResult::Detected { total_length, .. } => Ok(Some((true, total_length))),
+            ironrdp::rdclean_path::DetectionResult::NotEnoughBytes => Ok(None),
+            ironrdp::rdclean_path::DetectionResult::Failed => Err(ironrdp::core::other_err!(
+                "RDCleanPathHint",
+                "detection failed (invalid PDU)"
+            )),
+        }
+    }
+}
+
+#[diplomat::bridge]
+pub mod ffi {
+    use diplomat_runtime::DiplomatWriteable;
+    use ironrdp::rdclean_path::der::asn1::OctetString;
+    use std::fmt::Write;
+
+    use crate::error::ffi::{IronRdpError, IronRdpErrorKind};
+    use crate::error::ValueConsumedError;
+    use crate::utils::ffi::{OptionalString, VecU8, VecVecU8};
+
+    #[diplomat::opaque]
+    pub struct RdCleanPathPdu(pub Option<ironrdp::rdclean_path::RDCleanPathPdu>);
+
+    impl RdCleanPathPdu {
+        pub fn new_request(
+            x224_pdu: &VecU8,
+            destination: &str,
+            proxy_auth: &str,
+            pcb: &OptionalString,
```

Comment:
suggestion: Instead of creating this new "OptionalString" type, remove the parameter all together, and add setters and getters for all fields.